nss-day-cohort-46/rare-server-tragic-trolls
# Nc getting user posts by user

#Description

Added a get posts by user function to show the user only their posts.

#Testing Instructions

1. Add this data to your database

```
INSERT INTO Posts ('id', 'user_id', 'category_id', 'title', 'publication_date', 'image_url', 'content', 'approved') VALUES (Null, 1, 1, 'Title#1', '2021-04-02T23:57:07.374Z', 'https://res.cloudinary.com/nch66862/image/upload/v1617690859/man19_mfb18l.jpg', 'Post Text #1', 1);
INSERT INTO Posts ('id', 'user_id', 'category_id', 'title', 'publication_date', 'image_url', 'content', 'approved') VALUES (Null, 2, 1, 'Title#2', '2022-04-02T23:57:07.374Z', 'https://res.cloudinary.com/nch66862/image/upload/v1617690859/man19_mfb18l.jpg', 'Post Text #2', 0);
INSERT INTO Posts ('id', 'user_id', 'category_id', 'title', 'publication_date', 'image_url', 'content', 'approved') VALUES (Null, 1, 1, 'Title#3', '2023-04-02T23:57:07.374Z', 'https://res.cloudinary.com/nch66862/image/upload/v1617690859/man19_mfb18l.jpg', 'Post Text #3', 1);
INSERT INTO Posts ('id', 'user_id', 'category_id', 'title', 'publication_date', 'image_url', 'content', 'approved') VALUES (Null, 2, 1, 'Title#4', '2024-04-02T23:57:07.374Z', 'https://res.cloudinary.com/nch66862/image/upload/v1617690859/man19_mfb18l.jpg', 'Post Text #4', 0);
INSERT INTO Posts ('id', 'user_id', 'category_id', 'title', 'publication_date', 'image_url', 'content', 'approved') VALUES (Null, 1, 1, 'Title#5', '2025-04-02T23:57:07.374Z', 'https://res.cloudinary.com/nch66862/image/upload/v1617690859/man19_mfb18l.jpg', 'Post Text #5', 1);
INSERT INTO Posts ('id', 'user_id', 'category_id', 'title', 'publication_date', 'image_url', 'content', 'approved') VALUES (Null, 2, 1, 'Title#6', '2026-04-02T23:57:07.374Z', 'https://res.cloudinary.com/nch66862/image/upload/v1617690859/man19_mfb18l.jpg', 'Post Text #6', 0);
```

2. Ensure the debugger or watchgod is running
3. Type this into postman using a get method

```
http://localhost:8088/posts?userId=2
```

4. See that you got some data back
5. Type this into postman using a get method

```
http://localhost:8088/posts?userId=1
```

6. Ensure you get userId 1's data back
7. Approve this pull request
